explore/src/main/scala/explore/tabs/ObsTabTiles.scala

```diff
@@ -175,12 +170,12 @@ object ObsTabTiles:
       // Ags state
       .useStateView[AgsState](AgsState.Idle)
       // the configuration the user has selected from the spectroscopy modes table, if any
-      .useStateView(none[BasicConfigAndItc])
-      .useStateWithReuseBy: (props, _, _, _, selectedConfig) =>
-        itcQueryProps(props.observation.get, selectedConfig.get, props.allTargets)
+      .useStateView(none[InstrumentConfigAndItcResult])
+      .useStateWithReuseBy: (props, _, _, _, selectedConfig) => // itcQueryProps
+        ItcProps(props.observation.get, selectedConfig.get, props.obsTargets)
       .useState(Pot.pending[ItcAsterismGraphResults]) // itcGraphResults
       .useAsyncEffectWithDepsBy((props, _, _, _, selectedConfig, _, _) =>
-        itcQueryProps(props.observation.get, selectedConfig.get, props.allTargets)
+        ItcProps(props.observation.get, selectedConfig.get, props.obsTargets)
```

Comment:
Caan this be combined in one hook? Like one of those useResourceWhen...

Reply:
Let me check. At the very least, we can use a `localVal` so that we don't build `ItcProps` twice.

Reply:
I was able to simplify quite a bit by using `localVal` and `useEffectResult`. Thanks for the suggestion!